hr_infraction/tests/test_hr_infraction.py

```diff
@@ -0,0 +1,312 @@
+        _employee = self.employee if not employee else employee
+        return self.Infraction.with_user(_user).create(
+            {
+                "name": "Infraction #1",
+                "date": fields.Date.today(),
+                "employee_id": _employee.id,
+                "category_id": self.env.ref(
+                    "hr_infraction.infraction_category_care"
+                ).id,
+                "state": "draft",
+            }
+        )
+
+    def test_cant_remove_infraction_past_draft_state(self):
+        self.create_contract(
+            "draft", "normal", self.dstart, user=self.user_approve
+        ).signal_confirm()
+        infraction = self.create_infraction(user=self.user_entry)
+        self.assertEqual(infraction.state, "draft")
+        infraction.action_confirm()
+        with self.assertRaises(UserError):
+            infraction.unlink()
+
+    def test_infraction_termination_action(self):
+        self.create_contract(
+            "draft",
+            "normal",
+            self.dstart,
+            employee=self.user_approve.employee_id,
+            user=self.user_approve,
+        ).signal_confirm()
+        infraction = self.create_infraction(
+            self.user_approve.employee_id, user=self.user_entry
+        )
+        infraction.action_confirm()
+
+        self.assertEqual(infraction.state, "confirm")
+        self.assertEqual(infraction.employee_id.contract_id.state, "open")
+
+        self.InfActWizard.with_user(self.user_approve).with_context(
+            {"active_id": infraction.id}
+        ).create(
+            {"action_type": "dismissal", "effective_date": fields.Date.today()}
+        ).create_action()
+
+        self.assertEqual(infraction.state, "action")
+        self.assertEqual(infraction.employee_id.contract_id.state, "close")
+
+        for contract in infraction.employee_id.contract_ids:
+            self.assertEqual(contract.state, "close")
+
+    def test_infraction_transfer_action(self):
+        self.create_contract(
+            "draft",
+            "normal",
+            self.dstart,
+            employee=self.user_approve.employee_id,
+            user=self.user_approve,
+        ).signal_confirm()
+        infraction = self.create_infraction(
+            self.user_approve.employee_id, user=self.user_entry
+        )
+        infraction.action_confirm()
+
+        self.assertEqual(infraction.state, "confirm")
+        self.assertEqual(infraction.employee_id.contract_id.state, "open")
+
+        self.InfActWizard.with_user(self.user_approve).with_context(
+            {"active_id": infraction.id}
+        ).create(
+            {
+                "action_type": "transfer",
+                "new_job_id": self.job_junior.id,
+                "xfer_effective_date": (fields.Date.today() - timedelta(days=1)),
+            }
+        ).create_action()
+        transfer = self.Transfer.search([], limit=1)
+        self.assertFalse(len(transfer) == 0)
```

Comment:
This is not a good assert. This assert would pass even if the transfer failed but the module's demo data contained a demo transfer. Instead include terms in the search domain that would uniquely identify the transfer you created.